### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by aduplii**

### CODE REVIEW

File: app/config/config_test.yml [hunk 1]
```diff
@@ -14,3 +14,16 @@ web_profiler:
...
 swiftmailer:
     disable_delivery: true
+
+liip_functional_test: ~
+
+# Doctrine Configuration
+# Doctrine Configuration
+doctrine:
+    dbal:
+        driver: pdo_mysql
```
Comment (gregurco) [inline app/config/config_test.yml]: redefine here only `dbname`. It's no need to redefine driver, host, port and so on.

File: composer.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
         ]
     },
     "require": {
-        "php": ">=5.5.9",
+        "php": ">=7.1",
```
Comment (gregurco) [inline composer.json:24]: :+1:

File: tests/AppBundle/Controller/CategoryControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        $schemaTool = new SchemaTool($em);
+        $schemaTool->dropDatabase();
+        if (!empty($metadatas)) {
+            $schemaTool->createSchema($metadatas);
+        }
+        $this->postFixtureSetup();
+
+        $fixtures = [
+            \AppBundle\DataFixtures\CategoryFixture::class,
```
Comment (gregurco) [inline tests/AppBundle/Controller/CategoryControllerTest.php]: import classes

File: tests/AppBundle/Controller/CategoryControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+use AppBundle\Entity\Category;
+use AppBundle\Entity\Job;
+use Doctrine\ORM\Tools\SchemaTool;
+use Liip\FunctionalTestBundle\Test\WebTestCase;
+
+class CategoryControllerTest extends WebTestCase
+{
+    public function setUp()
+    {
+        $em = $this->getContainer()->get('doctrine')->getManager();
```
Comment (gregurco) [inline tests/AppBundle/Controller/CategoryControllerTest.php]: create tour own TestCase class and move this logic to separate method. It's common case and should be reusable but not written only in CategoryControllerTest class

File: tests/AppBundle/Controller/CategoryControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    public function testIndex()
+    {
+        $client = static::createClient();
+
+        // categories on homepage are clickable
+        $crawler = $client->request('GET', '/');
+        $link = $crawler->selectLink('Programming')->link();
+        $crawler = $client->click($link);
+        $this->assertEquals('AppBundle\Controller\CategoryController::showAction', $client->getRequest()->attributes->get('_controller'));
+        $this->assertTrue(200 === $client->getResponse()->getStatusCode());
```
Comment (gregurco) [inline tests/AppBundle/Controller/CategoryControllerTest.php]: use `assertEquals`

File: tests/AppBundle/Controller/JobControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        $this->assertEquals($job->getCompanySlug(), $client->getRequest()->attributes->get('company'));
+        $this->assertEquals($job->getLocationSlug(), $client->getRequest()->attributes->get('location'));
+        $this->assertEquals($job->getPositionSlug(), $client->getRequest()->attributes->get('position'));
+        $this->assertEquals($job->getId(), $client->getRequest()->attributes->get('id'));
+    }
+
+    /**
+     * Test nonexisting routes to job.
+     */
+    public function testShow()
```
Comment (gregurco) [inline tests/AppBundle/Controller/JobControllerTest.php]: separate this tests in two tests: testShowWithNotExistentJob and testShowWithEspiredJob
Better to write test with one test case

File: tests/AppBundle/Controller/JobControllerTest.php [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     */
+    protected function getExpiredJob()
+    {
+        $kernel = static::bootKernel();
+        $em = $kernel->getContainer()->get('doctrine.orm.entity_manager');
+
+        return $em->createQueryBuilder()
+            ->select('j')
+            ->from(\AppBundle\Entity\Job::class, 'j')
```
Comment (gregurco) [inline tests/AppBundle/Controller/JobControllerTest.php]: import class

File: tests/AppBundle/Utils/JobeetTest.php [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+namespace Tests\AppBundle\Utils;
+
+use AppBundle\Utils\Jobeet;
+use Symfony\Bundle\FrameworkBundle\Tests\TestCase;
+
+class JobeetTest extends TestCase
+{
+    public function testSlugify()
+    {
+        $this->assertEquals('okay-google', Jobeet::slugify('Okay Google'));
```
Comment (gregurco) [inline tests/AppBundle/Utils/JobeetTest.php]: try to rewrite this test with usage of `@dataProvider`

File: tests/AppBundle/AppTestTrait.php [hunk 9]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Tests\AppBundle;
+
+use Doctrine\ORM\Tools\SchemaTool;
+
+trait AppTestTrait
```
Comment (gregurco) [inline tests/AppBundle/AppTestTrait.php:7]: in this case it's not good practice to write trait. Main problems from my point view:
1) you use `$this->em` but what if trait will be included in class without `em`?
2) you use `postFixtureSetup` but what if trait will be included in class without this method?

File: tests/AppBundle/Utils/JobeetTest.php [hunk 10]
```diff
@@ -3,19 +3,36 @@
 namespace Tests\AppBundle\Utils;
 
 use AppBundle\Utils\Jobeet;
+use Doctrine\Common\Annotations\Annotation\IgnoreAnnotation;
 use Symfony\Bundle\FrameworkBundle\Tests\TestCase;
 
+/**
+ * @IgnoreAnnotation("dataProvider")
```
Comment (gregurco) [inline tests/AppBundle/Utils/JobeetTest.php]: why do you need `IgnoreAnnotation` here?

Comment (aduplii) [reply tests/AppBundle/Utils/JobeetTest.php]: I read about caveats with annotations here https://github.com/liip/LiipFunctionalTestBundle/tree/2.x#caveats
But I removed these annotations and tests still pass.

**Review by gregurco — APPROVED** [review]

**Review by gregurco — APPROVED** [review]

**Pull request merged**